tests/core/full_node/stores/test_coin_store.py

```diff
@@ -493,6 +495,169 @@ async def test_get_coin_states(db_version: int) -> None:
+            created_height = uint32(i % 10)
+            spent_height = uint32(created_height + 100)
+
+            puzzle_hash = std_hash(i.to_bytes(4, byteorder="big"))
+
+            coin = Coin(
+                std_hash(b"Parent Coin Id " + i.to_bytes(4, byteorder="big")),
+                puzzle_hash,
+                uint64(i),
+            )
+
+            if is_hinted:
+                hint = std_hash(b"Hinted " + puzzle_hash)
+                hints.append((coin.name(), hint))
+                puzzle_hashes.append(hint)
+            else:
+                puzzle_hashes.append(puzzle_hash)
+
+            coin_records.append(
+                CoinRecord(
+                    coin=coin,
+                    confirmed_block_index=created_height,
+                    spent_block_index=spent_height if is_spent else uint32(0),
+                    coinbase=False,
+                    timestamp=uint64(0),
+                )
+            )
+
+        # Initialize coin and hint stores.
+        coin_store = await CoinStore.create(db_wrapper)
+        hint_store = await HintStore.create(db_wrapper)
+
+        await coin_store._add_coin_records(coin_records)
+        await hint_store.add_hints(hints)
+
+        # Helper for syncing all of the coin states.
+        async def sync_states(*, include_spent: bool, include_unspent: bool, include_hinted: bool) -> List[CoinState]:
+            height = uint32(0)
+            all_coin_states: List[CoinState] = []
+            is_finished = False
+            remaining_phs = puzzle_hashes.copy()
+
+            while not is_finished:
+                (coin_states, height, is_finished) = await coin_store.batch_coin_states_by_puzzle_hashes(
+                    remaining_phs[:15000],
+                    min_height=height,
+                    include_spent=include_spent,
+                    include_unspent=include_unspent,
+                    include_hinted=include_hinted,
+                )
+                all_coin_states += coin_states
+
+                if is_finished:
+                    remaining_phs = remaining_phs[15000:]
+
+                    if len(remaining_phs) > 0:
+                        height = uint32(0)
+                        is_finished = False
+
+            return all_coin_states
+
+        # Make sure all of the coin states are found when batching.
+        all_coin_states = await sync_states(include_spent=True, include_unspent=True, include_hinted=True)
+        all_coin_states.sort(key=lambda cs: cs.coin.amount)
+
+        assert len(all_coin_states) == len(coin_records)
+
+        for i in range(min(len(coin_records), len(all_coin_states))):
+            assert coin_records[i].coin.name().hex() == all_coin_states[i].coin.name().hex(), i
+
+        # Make sure you can filter out hints.
+        all_coin_states = await sync_states(include_spent=True, include_unspent=True, include_hinted=False)
+        assert len(all_coin_states) == len(coin_records) - len(hints)
```

Comment:
I think it would be good to also ensure you get the exact records you expect. It's probably impractical to list every coin (among 15000), but you could ensure that all coins satisfies certain properties (like, not being spent or being spent, or having a hint or not having a specific hint).
Maybe ensure that all puzzle hashes are unique

Reply:
Made it more explicitly check the contents of the sorted output